endpoint: write header file on shutdown

squeed opened the pull request:
The endpoint's DNS history is preserved in the header file; we need it on restoration so IPs are preserved. This adds a shutdown hook in the EndpointManager to force dumping the header file to disk.

squeed committed:
endpoint: write header file on shutdown

The endpoint's DNS history is preserved in the header file; we need it
on restoration so IPs are preserved. This adds a shutdown hook in the
EndpointManager to force dumping the header file to disk.

Signed-off-by: Casey Callendrello <cdc@isovalent.com>

bimmlerd commented on pkg/endpointmanager/cell.go:
```diff
@@ -221,6 +221,7 @@ func newDefaultEndpointManager(p endpointManagerParams) endpointManagerOut {
 			},
 			OnStop: func(cell.HookContext) error {
 				cancel()
+				mgr.OnShutdown()
```
Do we really only want to do this `if EndpointGCInterval > 0`? I guess that's just mostly always true?

squeed committed:
test: FQDN: prevent names from being GCd when restarting

We rely on names not being eligible for garbage collection when
restarting. However, if the DNS name being tested has a low TTL, this
can trigger this accidentally.

Signed-off-by: Casey Callendrello <cdc@isovalent.com>
